tools/sof-tplgreader.py

```diff
@@ -125,6 +125,15 @@ def list_or(lst1, lst2):
                 output_lst.append(elem)
         return output_lst
 
+    @staticmethod
+    def list_diff(lst1, lst2):
+        assert(lst1 is not None and lst2 is not None)
+        output_lst = []
+        for elem in lst1:
+            if elem not in lst2:
+                output_lst.append(elem)
+        return output_lst
```

Comment:
Can input lists have duplicates? Does their order matter? If both answers are "no" then:
```python
return list(set(list1) - set(list2))
```

Reply:
@marc-hb Both answers are no, but the two lists used here are lists of dicts, they are not hashable, Python will report error on `set(list1)` . "unhahable type: dict"